tools/restatectl/src/commands/node/remove_nodes.rs

```diff
@@ -74,28 +61,36 @@ pub async fn remove_nodes(
     Ok(())
 }
 
-async fn create_metadata_client(
-    nodes_configuration: &NodesConfiguration,
-) -> anyhow::Result<MetadataStoreClient> {
-    // find metadata nodes
-    let addresses: Vec<_> = nodes_configuration
-        .iter_role(Role::MetadataServer)
-        .map(|(_, config)| config.address.clone())
-        .collect();
-    if addresses.is_empty() {
-        return Err(anyhow::anyhow!(
-            "No nodes are configured to run metadata-server role, this command only \
-             supports replicated metadata deployment"
-        ));
-    }
-
-    // todo make this work with other metadata client kinds as well; maybe proxy through Restate server
-    let metadata_store_client_options = MetadataClientOptions {
-        kind: MetadataClientKind::Replicated { addresses },
-        ..Default::default()
-    };
-
-    create_client(metadata_store_client_options)
+async fn update_nodes_configuration(
+    client: &MetadataStoreClient,
+    logs: &Logs,
+    opts: &RemoveNodesOpts,
+) -> Result<(), NodeOperationError> {
+    client
+        .read_modify_write::<_, _, anyhow::Error>(
+            NODES_CONFIG_KEY.clone(),
+            |nodes_configuration: Option<NodesConfiguration>| {
+                let mut nodes_configuration =
+                    nodes_configuration.context("Missing nodes configuration")?;
+
+                let disable_node_checker = DisableNodeChecker::new(&nodes_configuration, logs);
+
+                for node_id in &opts.nodes {
+                    disable_node_checker
+                        .safe_to_disable_node(*node_id)
+                        .context("It is not safe to disable node {node_id}")?;
+                }
+
+                for node_id in &opts.nodes {
+                    nodes_configuration.remove_node_unchecked(*node_id);
+                }
+
+                nodes_configuration.increment_version();
+
+                Ok(nodes_configuration)
+            },
+        )
         .await
-        .map_err(|e| anyhow::anyhow!("Failed to create metadata store client: {}", e))
+        .map(|_| ())
+        .map_err(Into::into)
```

Comment:
The reason why I would prefer to use only a conditional put here is because we kind of pin `Logs` across multiple retries. If logs is out of date, then we might come to the wrong conclusion that it is safe to remove a node. Using a conditional put will avoid the retries.

Reply:
Ah okay, sorry about that. I will revert it